### PR TITLE
Initial connection

### LINKED ISSUE #491: Implement Initial Connection Protocol
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

The initial connection should work with the game.

An initial connection is

1. Connect to server
2. Server replies if client already has an existing hero
3. If an existing hero exists the client receives the servers save
4. If an existing hero does not exist the client is taken to the character creation screen
5. After character creation the client receives the servers save
6. After the game loads time controls are re-enabled on the server

## Intended Design

https://drive.google.com/file/d/1KhiE62qUYYlH66fsh3BbWlmzo20UmGGS/view?usp=sharing

## Location

Create a new branch based off of development
There are various places in GameInterface and Coop.Core that will need to be updated.

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by garrettluskey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

Updated initial states.

## What is the new behaviour?

Resolves #491, #292

## Other information

### CODE REVIEW

File: source/Common.Tests/Common.Tests.csproj [hunk 1]
```diff
@@ -32,6 +34,9 @@
   </ItemGroup>
 
   <ItemGroup>
+    <Reference Include="LiteNetLib">
+      <HintPath>..\Coop.Tests\bin\Debug\LiteNetLib.dll</HintPath>
+    </Reference>
```
Comment (masesk) [inline source/Common.Tests/Common.Tests.csproj]: Why not use nuget here?

File: source/MissionTests/MissionTests.csproj [hunk 2]
```diff
@@ -42,17 +41,11 @@
...
-      <HintPath>..\packages\MSTest.TestFramework.2.2.7\lib\net45\Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions.dll</HintPath>
-    </Reference>
     <Reference Include="protobuf-net, Version=3.0.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
-      <HintPath>..\packages\protobuf-net.3.1.22\lib\net462\protobuf-net.dll</HintPath>
+      <HintPath>..\packages\protobuf-net.3.1.26\lib\net462\protobuf-net.dll</HintPath>
     </Reference>
     <Reference Include="protobuf-net.Core, Version=3.0.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
-      <HintPath>..\packages\protobuf-net.Core.3.1.22\lib\net462\protobuf-net.Core.dll</HintPath>
+      <HintPath>..\packages\protobuf-net.Core.3.1.26\lib\net462\protobuf-net.Core.dll</HintPath>
     </Reference>
```
Comment (masesk) [inline source/MissionTests/MissionTests.csproj:49]: Is this missing LiteNetLib?

File: source/Coop.Tests/Coop.Tests.csproj [hunk 3]
```diff
@@ -69,8 +69,8 @@
       <HintPath>..\packages\System.Diagnostics.DiagnosticSource.4.7.1\lib\net46\System.Diagnostics.DiagnosticSource.dll</HintPath>
     </Reference>
     <Reference Include="System.IO.Compression" />
-    <Reference Include="System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Memory.4.5.4\lib\net461\System.Memory.dll</HintPath>
+    <Reference Include="System.Memory, Version=4.0.1.2, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Memory.4.5.5\lib\net461\System.Memory.dll</HintPath>
```
Comment (masesk) [inline source/Coop.Tests/Coop.Tests.csproj:73]: Missing litenetlib?

Comment (garrettluskey) [reply source/Coop.Tests/Coop.Tests.csproj:73]: The tests themselves likely won't use LiteNetLib, unless it's not building for that reason this is a non-issue

File: source/MissionTests/MissionTests.csproj (hunk 2 above)
Comment (garrettluskey) [reply source/MissionTests/MissionTests.csproj:49]: MissionTests shouldn't have been touched in this branch but this change is a non-issue

File: source/Common.Tests/Common.Tests.csproj (hunk 1 above)
Comment (garrettluskey) [reply source/Common.Tests/Common.Tests.csproj:39]: Removed common tests as there were no tests

Comment (garrettluskey) [reply source/Common.Tests/Common.Tests.csproj]: Good catch fixed

**Review by EgardA — APPROVED** [review]
Looks good to me

**Pull request merged**